Clarify immutable actions help text (#18351)
github/codeql: actions/ql/src/Security/CWE-829/UnversionedImmutableAction.md

## Description

> This action is eligible for a new GitHub feature called Immutable Actions that is currently only available for internal users but will be publicly available soon. Immutable Actions are released as packages in the GitHub package registry instead of resolved from a pinned SHA at the repository. The immutable action provides the same immutability as pinning the version to a SHA but with improved readability.

Reviewer: How about something like this?

```suggestion
This action is eligible for Immutable Actions, a new GitHub feature that is currently only available for internal users. Immutable Actions are released as packages in the GitHub package registry instead of resolved from a pinned SHA at the repository. The Immutable Action provides the same immutability as pinning the version to a SHA but with improved readability and additional security guarantees.
```

Author: I like it
